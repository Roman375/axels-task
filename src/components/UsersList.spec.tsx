import { mount, shallow } from 'enzyme'
import { BrowserRouter as Router, Link } from 'react-router-dom'

import UserList from './UsersList'

describe('<UserList />', () => {
  const props = {
    employees: {
      employees: ['John', 'Ben'],
    },
    getOverviews: () => {},
    getEmployee: () => {},
  }

  describe('getEmployee is working', () => {
    const mockgetEmployee = jest.fn()
    
    const nextProps = {
      ...props,
      getEmployee: mockgetEmployee,
    }

    const userList = mount(
      <Router>
        <UserList {...nextProps} />
      </Router>
    )

    it('render properly', () => {
      expect(userList).toMatchSnapshot()
    })

    it('renders 2 employees', () => {
      expect(userList.find('Card')).toHaveLength(2)
    })

    expect(mockgetEmployee).toHaveBeenCalledTimes(1)
  })
})
