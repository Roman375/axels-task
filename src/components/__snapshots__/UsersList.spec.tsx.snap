// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserList /> getEmployee is working render properly 1`] = `ReactWrapper {}`;
